src/kakarot/instructions/environmental_information.cairo

```diff
@@ -487,42 +483,69 @@ namespace EnvironmentalInformation {
             contract_address=registry_address_, evm_contract_address=address_felt
         );
 
-        // handle case where there is no eth -> stark address mapping
-        if (starknet_contract_address == 0) {
-            return ctx;
-        }
 
-        // Get the bytecode from the Starknet_contract
-        let (bytecode_len, bytecode) = IEvmContract.bytecode(
-            contract_address=starknet_contract_address
-        );
-
-        // handle case were eth address returns no bytecode:
+        if (starknet_contract_address != 0) {
+            //  we get the bytecode from the Starknet_contract
+            let (bytecode_len, bytecode) = IEvmContract.bytecode(
+                contract_address=starknet_contract_address
+            ); 
+
+            tempvar syscall_ptr=syscall_ptr;
+            tempvar pedersen_ptr=pedersen_ptr;
+            tempvar range_check_ptr=range_check_ptr;                        
+            tempvar bitwise_ptr=bitwise_ptr;
+            tempvar bytecode_len=bytecode_len;
+        } else {
+            // otherwise handle case where there is no eth -> stark address mapping            
+            let bytecode : felt* = alloc();
+            let bytecode_len = 0;
+
+            tempvar syscall_ptr=syscall_ptr;
+            tempvar pedersen_ptr=pedersen_ptr;
+            tempvar range_check_ptr=range_check_ptr;                        
+            tempvar bitwise_ptr=bitwise_ptr;
+            tempvar bytecode_len=bytecode_len;            
 
-        if (bytecode_len == 0) {
-            return ctx;
         }
 
-        // TODO do we have the distinction between precompiles and warm and cold addresses?
+        tempvar syscall_ptr=syscall_ptr;
+        tempvar pedersen_ptr=pedersen_ptr;
+        tempvar range_check_ptr=range_check_ptr;                        
+        tempvar bitwise_ptr=bitwise_ptr;
+        tempvar bytecode_len=bytecode_len;            
```

Comment:
I believe so, yes. Otherwise there is a revoked reference error.

Reply:
because you redefine everything at the end of both branches so it shouldn't be

Reply:
I double checked and the only two tempvar assignments I needed to avoid reference revoke errors  here were `syscall_ptr` and `pedersen_ptr.` Updated to reflect this 👍